flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java

```diff
@@ -1553,6 +1697,95 @@ public void testTriggerAndConfirmSimpleSavepoint() throws Exception {
 		coord.shutdown(JobStatus.FINISHED);
 	}
 
+	@Test
+	public void testTriggerAndDeclineSimpleSavepointWithExecutionFailure() throws Exception {
```

Comment:
Do we need to add another test that `savepoint and cancel the job` and throw exception when doing savepoint

Reply:
According to the current implementation, savepoint will definitely not trigger the job to fail. Because savepoint is always forced (it will throw an exception and terminate subsequent execution). So this needs to be discussed.